app-misc/ranger/ranger-1.9.1.recipe

```diff
@@ -0,0 +1,93 @@
+SUMMARY="A simple, vim-like file manager"
+DESCRIPTION="Ranger is a console file manager that gives you greater \
+flexibility and a good overview of your files without having to leave your \
+console. It visualizes the directory tree in two dimensions: the directory \
+hierarchy on one, lists of files on the other, with a preview to the right so \
+you know where you'll be going."
+HOMEPAGE="https://ranger.github.io/"
+COPYRIGHT="2009-2018 Roman Zimbelmann
+	2010 David Barnett
+	2010 Lucas de Vries
+	2010 Sitaram Chamarty
+	2011 David Pugnasse
+	2011 ornicar
+	2011-2012 Abdó Roig-Maranges
+	2011-2012 M Rawash
+	2012 Serge Broslavsky
+	2012 joe
+	2013 Emanuel Guevel
+	2013 Joseph Tannhuber
+	2013-2014 GermainZ
+	2014 Célestin Matte
+	2014 Milan Svoboda
+	2014 rukai
+	2015 Alexander Buddenbrock
+	2015 Delisa Mason
+	2015 No Suck
+	2015 Randy Nance
+	2015 Ryan Burns
+	2015 anekos
+	2015 bastorran
+	2015-2017 nfnty
+	2015-2018 Wojciech Siewierski
+	2016-2018 Toon Nolten"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/$portName/$portName/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="7c32f96080030866b7b95740c40b17a37d6ee86055e054e9e0759e79e2bf6b9a"
+SOURCE_FILENAME="$portVersionedName.tar.gz"
+
+ARCHITECTURES="any"
+
+PROVIDES="
+	$portName = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+
+PYTHON_PACKAGES=(python python3)
+PYTHON_VERSIONS=(2.7 3.6)
+for i in "${!PYTHON_PACKAGES[@]}"; do
+pythonPackage=${PYTHON_PACKAGES[i]}
+pythonVersion=${PYTHON_VERSIONS[$i]}
+eval "PROVIDES_${pythonPackage}=\"\
+	${portName}_$pythonPackage = $portVersion\n\
+	cmd:ranger = $portVersion\n\
+	cmd:rifle = $portVersion\
```

Comment:
A problem here is that ranger provides some `cmd:`s. It's currently possible to install both `python3` and `python` version, and the one installed later(?) will have it's `cmd:`s take priority. Normally this should be fine, but since ranger is capable of loading python plugins, I think we should either:

- Only package the `cmd:`s for a specific python version. I would vote for python2 in this case, since it works equally well for all architectures
- Further splitting the package with `_python_cmd` and `_python3_cmd`

Reply:
With a few `cmd:`-style conflcts we can fix this. Here are the changes, with a few cosmetic neutral changes too :)
```diff
 for i in "${!PYTHON_PACKAGES[@]}"; do
 pythonPackage=${PYTHON_PACKAGES[i]}
 pythonVersion=${PYTHON_VERSIONS[$i]}
-eval "PROVIDES_${pythonPackage}=\"\
-	${portName}_$pythonPackage = $portVersion\n\
-	cmd:ranger = $portVersion\n\
-	cmd:rifle = $portVersion\
-	\"; \
-REQUIRES_$pythonPackage=\"\
-	haiku\n\
-	cmd:python$pythonVersion\
-	\""
+eval "\
+	PROVIDES_$pythonPackage=\"\n\
+		${portName}_$pythonPackage = $portVersion\n\
+		cmd:ranger = $portVersion\n\
+		cmd:rifle = $portVersion\n\
+		\"; \
+	REQUIRES_$pythonPackage=\"\n\
+		haiku\n\
+		cmd:python$pythonVersion\n\
+		\"; \
+	CONFLICTS_$pythonPackage=\"\n\
+		cmd:ranger\n\
+		cmd:rifle\n\
+		\"; \
+	"
 BUILD_REQUIRES="$BUILD_REQUIRES
 	setuptools_$pythonPackage"
```
(I've just tested and it works fine.)
What do you think?
(

Reply:
`ranger` could be used as a python module, so I don't think we should not forbid one from coding with `python3` while running `ranger` with `python2` plugins. But I don't think anyone would have that use case anyway, so this seem fair enough to me.

As of styling, how about this?

```sh
eval "PROVIDES_$pythonPackage='
	${portName}_$pythonPackage = $portVersion
	cmd:ranger = $portVersion
	cmd:rifle = $portVersion
	'"
eval "REQUIRES_$pythonPackage='
	haiku
	cmd:python$pythonVersion
	'"
eval "CONFLICTS_$pythonPackage='
	cmd:ranger
	cmd:rifle
	'"

Reply:
That's a good idea, for the style.

For the `CONFLICTS` I must say that I'm not as confortable with my own suggestion now that you have broght to my attention that it would not allow some use cases.
The most flexible solution would be to have the conflicts for the _python_cmd, and allow the modules of each python subpackage to be co-installed.
But this would mean that we would have one base package and 4 subpackages...
I'm OK with this but I guess some users would prefer the current layout.

Shall we keep this PR open for 1 or 2 days, so that we can all think about it and see what @tts2k and the other contributors think about this?

Hmm, we might workaround the flexibility issue by switching to a layout similar to the one we have for youtube-dl. But don't think this would be worth for this particular package.

Reply:
Personally I think that we should sacrifice some flexibility for users to make maintainance easier. 

After a few thoughts, I believe your suggestion is pretty good as the usecase I mentioned probably never exists. Most Linux distros only package for a specific version of python, and I haven't seen anyone complaining about that yet. Also, if one really needs those use cases, they should be capable of modifying our recipe, or they could just convert whatever plugin they're using to python3 anyway.

Reply:
Yeah, I guess you are right, @alaviss. Let's go for your style of the conflicts I suggested.
@tts2k, what do you think?

Reply:
I'm ok with @alaviss style.